### ISSUE TITLE
Can't change audio tracks and subtitles in certain videos in LG webOS app

### ISSUE THREAD

**Issue by Goulmort** [Stremio/stremio-bugs#792]
**Describe the bug**
Basically can't change audio tracks and subtitles from certain sources in LG's native webOS app. For certain videos If **same video source** have multiple audio tracks i can't change audio on webOS stremio app, but i can on android app.  In some cases audio won't even play, just the video.  From addons i'm using Torrentio,  Opensubtitles, CyberFlix: The Catalog Scrapper. Also i'm using RD.
**To Reproduce**
1. Open Stremio native app on LG TV
2. Navigate to The X Files tv show
3. Select first episode from the first season (1. Pilot)
4. Select to play first highest quality video source (file size 11.15GB, source Rutracker, it has embedded subtitles, and GB/RU/UA audio tracks)
5. Video starts playing with russian voiceovers and no subitles. Can't change audio track, and can't select subtitles since there are  none.
6. Doing same steps on android app i can change audio tracks and embedded subititles

**Expected behavior**
I expect in lower left corner to have an option to change embedded audio track, and choose appropriate subtitles. Instead you just have 3 options in bottom left corner, subtitle select (can't choose any subtitle btw since there is none for some reason), playback speed and options to select different episodes from season to play.

**Screenshots**
I've provided two pictures, one from webOS version with the missing option for audio select, and second one as screenshot from android app from that same video source where i can change audio and subtitles.
(https://postimg.cc/gallery/L9NJqvd)

**Smartphone (please complete the following information):**
 - Device: OnePlus 12
 - OS: OxygenOS 14.0 / Android 14

**TV (please complete the following information):**
 - Device: LG C2 oled (OLED42C21LA)
 - OS: webOS23
 - Version 13.30.56

**Additional context**
Stremio app version on TV is Theater v 1.3.0
Stremio app version on Android phone is 1.6.12

**Comment by Goulmort** [comment]
Same exact behaviour can be observed for example on movie **Oppenheimer**. Navigate to the highest quality video source, that would be Rutracker again, file size 97.6GB, audio tracks from GB/RU/UA. This time video is with Dolby Vision | HDR. Same thing happens, video starts playing, audio is dubbed in russian language and there is no option to select different audio tracks. No embedded subtitles also.

**Comment by py4747** [comment]
I've also experienced the same issue on several files, mostly large format files. Hoping it gets fixed

**Comment by KetanM94** [comment]
@Goulmort @py4747 
I faced the same issue and found a workaround.
This is how I do it:
Play the video and press back immediately. Then play the same video again. Now you'll see the option to change audio tracks.
I tested with 3 different videos and it worked as I described above.
It may or may not work for you. Please test this and see if it works for you.

**Comment by py4747** [comment]
@Goulmort Tried. Didn't work. Maybe you can record a video and show how it works. But I tried and didn't see results unfortunetly

**Comment by KetanM94** [comment]
> @Goulmort Tried. Didn't work. Maybe you can record a video and show how it works. But I tried and didn't see results unfortunetly 

https://github.com/Stremio/stremio-bugs/assets/17568553/6676240c-024d-4465-8df7-d340c788f741

As you can see in video, I play the stream. Then quickly press back. Then play it again. You can see the speaker/audio icon appears. Then I select tracks with cursor. Sometimes you can't navigate to that icon as seen in the video. Just redoing the above process can make it work correctly or just use magic remote cursor to select the icon.

**Comment by py4747** [comment]
@KetanM94 I think I know what's going on. seems like this works with some files ( possibly the ones that show the dual audio and the flag in the selection menu), but if I go for a large size file of that  same movie which doesn't show the different flags the audio just autos to the commentry track and your method doesn't fix it. Still good to know for some files, but hope it gets fixed soon becuase I really like the larger files.

**Comment by KetanM94** [comment]
> @KetanM94 I think I know what's going on. seems like this works with some files ( possibly the ones that show the dual audio and the flag in the selection menu), but if I go for a large size file of that  same movie which doesn't show the different flags the audio just autos to the commentry track and your method doesn't fix it. Still good to know for some files, but hope it gets fixed soon becuase I really like the larger files.

Yes! You are right. This doesn't work with larger files.

**Comment by Bran89** [comment]
Any update on this being resolved eventually?

**Comment by Goulmort** [comment]
> Any update on this being resolved eventually?

Lead dev Jaruba mentioned on reddit that this issue should be addressed in next stremio update scheduled for webOS. Keep in mind app update for LG tv's need more time to be approved by LG before it goes live.

**Comment by onfire-d** [comment]
> > Any update on this being resolved eventually?
> 
> Lead dev Jaruba mentioned on reddit that this issue should be addressed in next stremio update scheduled for webOS. Keep in mind app update for LG tv's need more time to be approved by LG before it goes live.

That’s great to hear, is there, however, any ETA?

**Comment by dhrumilpatel30** [comment]
Any updates in regards to this issue? It is understandable that the time is taken due to LG marketplace approval is longer. If you need any contributions or man power I would love to take part.

**Comment by dhrumilpatel30** [comment]
Ohh!! It is sad. Thanks for the reply.

**Comment by tomboiscomputers** [comment]
Any Updates since this? or work arounds

**Comment by Goulmort** [comment]
> Any Updates since this? or work arounds

Afaik, devs are already finished update for Stremio on webOS but LG has notoriously difficult and long update approval system, so we may be in for a while. It's really mind-boggling that you need to wait this long to push an update. 

Also, no workarounds around this for Stremio on webOS. There is an alternative to fix all of this, but since this is Stremio git i will not mention it here. I'll pm you instead.

**Comment by 29sohil89** [comment]
> 
> Also, no workarounds around this for Stremio on webOS. There is an alternative to fix all of this, but since this is Stremio git i will not mention it here. Just search Stremio alternative on webOS and work from there.


Can you please tell me the alternative or any reddit post mentioning the workaround for the same 

I hv the lg smart monitor with webos 22

Everything works fine and smooth except for audio selection

**Comment by anderdl18** [comment]
> > Also, no workarounds around this for Stremio on webOS. There is an alternative to fix all of this, but since this is Stremio git i will not mention it here. Just search Stremio alternative on webOS and work from there.
> 
> Can you please tell me the alternative or any reddit post mentioning the workaround for the same
> 
> I hv the lg smart monitor with webos 22
> 
> Everything works fine and smooth except for audio selection

We are waiting for LG's goodwill to release the update that, to date, has not been released. The solution may be to use the web player, but I tested it and the content is like unsupported video.

**Comment by Goulmort** [comment]
> > > Also, no workarounds around this for Stremio on webOS. There is an alternative to fix all of this, but since this is Stremio git i will not mention it here. Just search Stremio alternative on webOS and work from there.
> > 
> > 
> > Can you please tell me the alternative or any reddit post mentioning the workaround for the same
> > I hv the lg smart monitor with webos 22
> > Everything works fine and smooth except for audio selection
> 
> We are waiting for LG's goodwill to release the update that, to date, has not been released. The solution may be to use the web player, but I tested it and the content is like unsupported video.

Solution is to sideload entirely new app into webOS. [This](https://tinyurl.com/2s4jc498) is a setup guide. How to install look for YT videos . It will require creating dev account at lg developers website.

**Comment by 29sohil89** [comment]
Thanks @Goulmort 

Looks like i am gonna wait for the official stremio update for this specific fix 

I hope it will come to webos 22 as well

**Comment by TRtomasz** [comment]
New update has been released

**Comment by Goulmort** [comment]
With new update everything works as intended. Big props to the devs! If you still have no sound on C2/maybe G2 models or maybe you have only dubbed sound or commentary track  you can thanks LG for that since they did not licensed DTS-HD for that tv's, and nowadays many high quality files use that audio format. Solution for that is posted couple posts above. Considered this issue CLOSED.

**Comment by ablott976** [comment]
Hi, I al still experiencing this issue on the latest update for LG, as explained

**Comment by Mariano228** [comment]
Yeah after the update, most torrents play without ANY audio track at all! And there's not an icon to change it . So the update actually made it worse. Am i missing something?

**Comment by anderdl18** [comment]
> Yeah after the update, most torrents play without ANY audio track at all! And there's not an icon to change it . So the update actually made it worse. Am i missing something?

Yes, I had this same problem yesterday when trying to watch Rogue One and I've had the problem with some other productions that I won't remember now.

**Comment by Goulmort** [comment]
I don't know what models other users have, on my c2 stremio works fine. Every file/source with dts-hd audio will not play sound, or you will get only commentary track that you can hear. Other that that it works as intended.

However i'm not on the latest os version, i'm on 13.30.80. 

What people can do is to hard restart their tv's. Turn off tv, pull the plug from the wall, wait 5-10 minutes, press "turn on" button on the tv (it's usually located at bottom of the tv in the middle), hold it for 30 second (tv is still not connected to the power), then release it, put the plug into the wall, start the tv from the remote. Hope it clears some issues. But don't expect to hear sound in all video files because big portion of 4K and some 1080p files uses dts-hd audio that LG didn't licensed for some models.

**Comment by Mariano228** [comment]
> I don't know what models other users have, on my c2 stremio works fine. Every file/source with dts-hd audio will not play sound, or you will get only commentary track that you can hear. Other that that it works as intended.
> 
> However i'm not on the latest os version, i'm on 13.30.80.
> 
> What people can do is to hard restart their tv's. Turn off tv, pull the plug from the wall, wait 5-10 minutes, press "turn on" button on the tv (it's usually located at bottom of the tv in the middle), hold it for 30 second (tv is still not connected to the power), then release it, put the plug into the wall, start the tv from the remote. Hope it clears some issues. But don't expect to hear sound in all video files because big portion of 4K and some 1080p files use dts-hd audio that LG didn't licensed for some models.

Thanks for the tip. So are you telling me that if you play open stremio native app on your C2 and play the 2015 documentary "Ronaldo" (6.56GB size) you can hear the audio?

**Comment by Mariano228** [comment]
> > I don't know what models other users have, on my c2 stremio works fine. Every file/source with dts-hd audio will not play sound, or you will get only commentary track that you can hear. Other that that it works as intended.
> > However i'm not on the latest os version, i'm on 13.30.80.
> > What people can do is to hard restart their tv's. Turn off tv, pull the plug from the wall, wait 5-10 minutes, press "turn on" button on the tv (it's usually located at bottom of the tv in the middle), hold it for 30 second (tv is still not connected to the power), then release it, put the plug into the wall, start the tv from the remote. Hope it clears some issues. But don't expect to hear sound in all video files because big portion of 4K and some 1080p files use dts-hd audio that LG didn't licensed for some models.
> 
> Thanks for the tip. So are you telling me that if you open the stremio native app on your C2 and play the 2015 documentary "Ronaldo" (6.56GB size) you can hear the audio? And you can see the audio menu icon (the one with a speaker)?

@Goulmort

**Comment by Goulmort** [comment]
> > > I don't know what models other users have, on my c2 stremio works fine. Every file/source with dts-hd audio will not play sound, or you will get only commentary track that you can hear. Other that that it works as intended.
> > > However i'm not on the latest os version, i'm on 13.30.80.
> > > What people can do is to hard restart their tv's. Turn off tv, pull the plug from the wall, wait 5-10 minutes, press "turn on" button on the tv (it's usually located at bottom of the tv in the middle), hold it for 30 second (tv is still not connected to the power), then release it, put the plug into the wall, start the tv from the remote. Hope it clears some issues. But don't expect to hear sound in all video files because big portion of 4K and some 1080p files use dts-hd audio that LG didn't licensed for some models.
> > 
> > 
> > Thanks for the tip. So are you telling me that if you open the stremio native app on your C2 and play the 2015 documentary "Ronaldo" (6.56GB size) you can hear the audio? And you can see the audio menu icon (the one with a speaker)?
> 
> [@Goulmort](https://github.com/Goulmort)

@Mariano228 Sorry for late reply. I don't see the icon, and there is no sound (file has dts-hd as audio format). I see that file is x264 from ROVERS group. Most of their releases use dts-hd for audio, so i tend to avoid them if possible, at least on Stremio app.

**Comment by Mariano228** [comment]
> > > > I don't know what models other users have, on my c2 stremio works fine. Every file/source with dts-hd audio will not play sound, or you will get only commentary track that you can hear. Other that that it works as intended.
> > > > However i'm not on the latest os version, i'm on 13.30.80.
> > > > What people can do is to hard restart their tv's. Turn off tv, pull the plug from the wall, wait 5-10 minutes, press "turn on" button on the tv (it's usually located at bottom of the tv in the middle), hold it for 30 second (tv is still not connected to the power), then release it, put the plug into the wall, start the tv from the remote. Hope it clears some issues. But don't expect to hear sound in all video files because big portion of 4K and some 1080p files use dts-hd audio that LG didn't licensed for some models.
> > > 
> > > 
> > > Thanks for the tip. So are you telling me that if you open the stremio native app on your C2 and play the 2015 documentary "Ronaldo" (6.56GB size) you can hear the audio? And you can see the audio menu icon (the one with a speaker)?
> > 
> > 
> > [@Goulmort](https://github.com/Goulmort)
> 
> [@Mariano228](https://github.com/Mariano228) Sorry for late reply. I don't see the icon, and there is no sound (file has dts-hd as audio format). I see that file is x264 from ROVERS group. Most of their releases use dts-hd for audio, so i tend to avoid them if possible, at least on Stremio app.

Thanks for checking. Why don’t you just use Stremio through a firestick so you won’t have to avoid certain files? Certain titles may not at all be available apart from dts-hd

**Comment by Goulmort** [comment]
@Mariano228 while getting a firestick or cube would be a prefered and easier solution, sideloading an [app](https://archive.is/03ZZM) in webOS worked surprisingly well. I can play any file i want, so if i want to play let's say...DV .mkv file that Stremio can't play, and that file has also dts-hd audio that Stremio can't play on my tv i just switch to another app on tv and it's all good. 

Point being, i still prefer native apps than entirely new streaming device. Seen some post about them struggling to play high bitrate remux files and this setup with tv works perfectly well. Keep in mind another reason for me using this second app is DV files in .mkv format. You can't play them on any LG tv. I mean you can, but it will switch to HDR mode instead of DV mode. Only solution for that is another streaming device or...another app.

**Comment by kvenn** [comment]
Not sure if this is the correct place for this, but on `web.stremio.com` most of my videos don't have audio playing and there is no way to select an audio track. On the MacOS app, the audio is correct and I can select the audio track for the the same file.

**Comment by Mariano228** [comment]
> [@Mariano228](https://github.com/Mariano228) while getting a firestick or cube would be a prefered and easier solution, sideloading an [app](https://archive.is/03ZZM) in webOS worked surprisingly well. I can play any file i want, so if i want to play let's say...DV .mkv file that Stremio can't play, and that file has also dts-hd audio that Stremio can't play on my tv i just switch to another app on tv and it's all good.
> 
> Point being, i still prefer native apps than entirely new streaming device. Seen some post about them struggling to play high bitrate remux files and this setup with tv works perfectly well. Keep in mind another reason for me using this second app is DV files in .mkv format. You can't play them on any LG tv. I mean you can, but it will switch to HDR mode instead of DV mode. Only solution for that is another streaming device or...another app.

You call this convenient but , Respectfully , i find it not really convenient . If you knew in advance which files will play in Stremio then you could open Stremio for those ones and the other app for the ones That you know won’t work I. Stremio. But since you don’t know which ones have dts audio, then your setup is rather inconvenient.

Say you want to watch the Ronaldo documentary mentioned above. You’d have to open stremio , look for the file , play it , only to realise it won’t play. Then you have to go open the other app. 

Instead , if you play Stremio through fire stick or cube , you can watch all files on Stremio without issues and that’s way more convenient

**Comment by py4747** [comment]
#  Instead , if you play Stremio through fire stick or cube , you can watch any files on Stremio without encountering any issues, and that’s way more convenient

@Mariano228 I'm not the person you've replied to, but this definitely isn't the case with the Firestick. It black screens on so many DV content. I don't know about the Cube since I only have the Firestick 4K MAX, and it's been pretty useless. The LG Stremio app feels a lot easier to get to and use on the TV, and has similar issues with DV content, but at least it launches faster. I don't recommend the Firestick.

**Comment by Mariano228** [comment]
> > Instead , if you play Stremio through fire stick or cube , you can watch any files on Stremio without encountering any issues, and that’s way more convenient
> 
> [@Mariano228](https://github.com/Mariano228) I'm not the person you've replied to, but this definitely isn't the case with the Firestick. It black screens on so many DV content. I don't know about the Cube since I only have the Firestick 4K MAX, and it's been pretty useless. The LG Stremio app feels a lot easier to get to and use on the TV, and has similar issues with DV content, but at least it launches faster. I don't recommend the Firestick.

I've got a cube. Just for reference, not once have I ever tried to open a file in Stremio that wouldn't play because of the DV

**Comment by jackyagami-alt** [comment]
Why this issue closed because i have been experiencing the same problem on my Lg Tv Native App (Latest App update)

**Comment by anderdl18** [comment]
> Why this issue closed because i have been experiencing the same problem on my Lg Tv Native App (Latest App update)

I have already accepted that this problem will probably never be resolved, on both LG and Samsung. The internal player on Tizen and webOS is terrible compared to Google TV's. I've already researched this and there is no solution: either switch to a Smart TV with Google TV, or buy an official Box that has Google TV, like the Xiaomi Mi Box or an Onn 4K. LG's internal player doesn't support heavy files, and when they contain multiple audio tracks, it simply ignores those audios. My next television will most definitely be a TCL with Google TV. This problem with Stremio on my LG has reached a point where it's simply no longer acceptable.

**Comment by py4747** [comment]
Just to add to what @anderdl18 said. If anyone decided to buy an external device, the internet will tell you to buy an Amazon firestick. Stay absolutely away from the Firestick. I bought the Firestick 4k max and its absolutely garbage tier when it comes to file support, lots of larger files and formats just black screen. Additionally it has forced Amazon ads that cannot be turned off, it forces you to stare at the Amazon home screen every time or watch a trailer and increases your startup wait.

**Comment by jackyagami-alt** [comment]
> Just to add to what [@anderdl18](https://github.com/anderdl18) said. If anyone decided to buy an external device, the internet will tell you to buy an Amazon firestick. Stay absolutely away from the Firestick. I bought the Firestick 4k max and its absolutely garbage tier when it comes to file support, lots of larger files and formats just black screen. Additionally it has forced Amazon ads that cannot be turned off, it forces you to stare at the Amazon home screen every time or watch a trailer and increases your startup wa

The only option i have left is Google Streamer if i don't go for firestick. These 2 are the only options available in my region

**Comment by yeiddi** [comment]
> > Why this issue closed because i have been experiencing the same problem on my Lg Tv Native App (Latest App update)
> 
> I have already accepted that this problem will probably never be resolved, on both LG and Samsung. The internal player on Tizen and webOS is terrible compared to Google TV's. I've already researched this and there is no solution: either switch to a Smart TV with Google TV, or buy an official Box that has Google TV, like the Xiaomi Mi Box or an Onn 4K. LG's internal player doesn't support heavy files, and when they contain multiple audio tracks, it simply ignores those audios. My next television will most definitely be a TCL with Google TV. This problem with Stremio on my LG has reached a point where it's simply no longer acceptable.


It was working fine all year on my LG even with super big files and it stopped as of 3 weeks ago it starts working for a week and then it randomly stops

**Comment by anderdl18** [comment]
> > > Why this issue closed because i have been experiencing the same problem on my Lg Tv Native App (Latest App update)
> > 
> > 
> > I have already accepted that this problem will probably never be resolved, on both LG and Samsung. The internal player on Tizen and webOS is terrible compared to Google TV's. I've already researched this and there is no solution: either switch to a Smart TV with Google TV, or buy an official Box that has Google TV, like the Xiaomi Mi Box or an Onn 4K. LG's internal player doesn't support heavy files, and when they contain multiple audio tracks, it simply ignores those audios. My next television will most definitely be a TCL with Google TV. This problem with Stremio on my LG has reached a point where it's simply no longer acceptable.
> 
> It was working fine all year on my LG even with super big files and it stopped as of 3 weeks ago it starts working for a week and then it randomly stops

I only have issues with extremely heavy files on Stremio for LG if the movies have more than one audio track. If there are up to two tracks, it still shows them, but if it’s a REMUX file with several audio tracks, it always ignores all of them and usually loads only the director’s commentary, if available. If not, it only shows the English audio and completely ignores the others. I’ve had to skip many movies in my own language because of this or watch lower-quality versions that included it. I don't mind watching subtitled movies at all, but there are some I much prefer to watch in my language because I appreciate and like the voice actors' work.

**Comment by nenad** [comment]
OK, I found a solution: https://github.com/Stremio/stremio-bugs/issues/2158

tl;dr is to disable IPv6 on your TV and this will work. Settings -> Network -> Uncheck "Enable IPv6".

I have no idea how this ties back to the stream not being able to select audio files because of this.

**Comment by yeiddi** [comment]
Yep this is what solved it for me a few months ago as well, for some reason IPv6 does weird things. Even for gaming. I posted the solution on reddit but forgot to reply here.
